### PR TITLE
Add static analysis for a single file

### PR DESCRIPTION
**Pull request by ZhiHanL**

### CODE REVIEW

File: src/main/java/com/github/algobot76/surabaya/SurabayaAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package com.github.algobot76.surabaya;
+
+import com.github.algobot76.surabaya.util.Project;
+import com.github.algobot76.surabaya.util.StaticAnalysisVisitor;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.visitor.VoidVisitor;
+import com.github.javaparser.utils.CodeGenerationUtils;
+import com.github.javaparser.utils.SourceRoot;
+
+public class SurabayaAnalyzer {
```
Comment (algobot76) [inline src/main/java/com/github/algobot76/surabaya/SurabayaAnalyzer.java:10]: I don't think this file should be at the same directory level as `SurayabaApplication`. Maybe we should put it in `util` package. Also, the name might be a bit too long. How about renaming it as `Analyzer`?

File: src/main/java/com/github/algobot76/surabaya/SurabayaApplication.java [hunk 2]
```diff
@@ -2,34 +2,21 @@
...
-		// working
-		SourceRoot sourceRoot = new SourceRoot(
-				CodeGenerationUtils.mavenModuleRoot(SurabayaApplication.class).resolve("src/test/testJavaProject/src"));
-
-		CompilationUnit cu = sourceRoot.parse("ast", "DEC.java");
-		System.out.printf("Test: %s", cu.getChildNodes().get(0).toString());
+		SurabayaAnalyzer analyzer = new SurabayaAnalyzer();
+		Project parsedProject = analyzer.analyze("src/test/testProject/src");
+		ObjectMapper mapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+		System.out.println(mapper.writeValueAsString(parsedProject));
```
Comment (algobot76) [inline src/main/java/com/github/algobot76/surabaya/SurabayaApplication.java:19]: I think a better way to handle the exception is below:

```java
try {
    System.out.println(mapper.writeValueAsString(parsedProject));
} catch (JsonProcessingException e) {
    System.out.println(e.getMessage());
}
```

File: src/main/java/com/github/algobot76/surabaya/util/JavaFile.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+@JsonIgnoreProperties(value = { "currentClass" })
+@Data
+public class JavaFile {
+
+	private Klass currentClass = null;
```
Comment (algobot76) [inline src/main/java/com/github/algobot76/surabaya/util/JavaFile.java:14]: Personally, I prefer to add `@JsonIgnore` directly on the field `currentClass`.

**Comment by algobot76** [comment]
And I think the `resources directory` is a better place for `testJavaProject` because `testJavaProject` is not our unit test files but input files for the analyzer.

**Comment by ZhiHanL** [comment]
JSON output from tinyVars main file:
```
{
  "packages" : {
    "ui" : {
      "files" : [ {
        "imports" : [ "PROGRAM", "EXP", "Tokenizer", "Arrays", "HashMap", "List", "Map" ],
        "classes" : [ {
          "name" : "Main",
          "type" : "Class",
          "fields" : [ {
            "name" : "symbolTable",
            "type" : "Map<String, Integer>",
            "access_modifier" : "public"
          } ],
          "methods" : [ {
            "name" : "main",
            "parameters" : [ {
              "name" : "args",
              "type" : "String[]"
            } ],
            "access_modifier" : "public",
            "return_type" : "void"
          } ],
          "constructors" : [ ],
          "access_modifier" : "public",
          "line_count" : 14
        } ]
      } ]
    }
  }
}
```

**Pull request merged**